src/components/VcModal/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import { DEFAULT_COLORS, WEBSITE } from '../../utils/constants';
+import '../styles.css';
+import QrCode from '../QrCode';
+import GateKeeperLogo from '../../assets/logotext.png';
+import {
+  PolygonAuthorizationResponse,
+  SocketMessage,
+  VcModalProps,
+} from '../../types';
+import Container from '../Container';
+import { doRoleCheck } from '../../utils/backendCalls';
+
+const VcModal = ({
+  customization,
+  response,
+  roles,
+  shouldGetDID,
+  closeModal,
+}: VcModalProps) => {
+  const { backgroundColor, textColor } = customization;
+  const [showQrCode, setShowQrCode] = useState(shouldGetDID);
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [vcIdsArray, setVcIdsArray] = useState<string[]>([]);
+
+  const handleSocketResponse = async (msg: SocketMessage) => {
+    if (msg?.error) {
+      setError(msg.error);
+      return;
+    }
+
+    if (msg.success) {
+      setShowQrCode(false);
+      setIsLoading(true);
+
+      setTimeout(async () => {
+        if (roles) {
+          const res = await Promise.all(
+            roles.map(async role => doRoleCheck(role, false))
+          );
+          if (res) {
+            const vcIdsArray = res.map(res => res.vcs[0]);
+            setVcIdsArray(vcIdsArray);
+            setIsLoading(false);
+          }
+        }
+      }, 5000);
```

Comment:
@KyrneDev this is the problem with the VC ids after do the auth stuff. When i get the success msg from the socket the i have to fetch the VC ids pending calling to doRoleCheck route. For some reason it seem that happend to fast and im no able to get that ids from the call. Any solution on your mind?

Reply:
So to confirm, the issue is that your call to doRoleCheck is too quick after the auth and the VCids aren't showing up? I would double check this is still true, because the user.save() happens before the socket message gets sent back to you.

Reply:
yes, thats the issue.
Can you try so you can canfirm this.